test/next_ls/commands/alias_test.exs

```diff
@@ -0,0 +1,197 @@
+defmodule NextLS.Commands.AliasTest do
+  use ExUnit.Case, async: true
+
+  alias GenLSP.Structures.TextEdit
+  alias GenLSP.Structures.WorkspaceEdit
+  alias NextLS.Commands.Alias
+
+  # @parse_error_code -32_700
+
+  describe "alias-refactor" do
+    test "works with single calls" do
+      uri = "my_app.ex"
+
+      text =
+        String.split(
+          """
+          defmodule MyApp do
+            def to_list(map) do
+              Foo.Bar.to_list(map)
+            end
+          end
+          """,
+          "\n"
+        )
+
+      expected_edit =
+        String.trim("""
+        defmodule MyApp do
+          alias Foo.Bar
+
+          def to_list(map) do
+            Bar.to_list(map)
+          end
+        end
+        """)
+
+      line = 2
+      position = %{"line" => line, "character" => 6}
+
+      assert %WorkspaceEdit{changes: %{^uri => [edit = %TextEdit{range: range}]}} =
+               Alias.refactor(%{uri: uri, text: text, position: position})
+
+      assert edit.new_text == expected_edit
+      assert range.start.line == 0
+      assert range.start.character == 0
+      assert range.end.line == 4
+      assert range.end.character == 3
+    end
+
+    test "works with multiple def calls" do
+      uri = "my_app.ex"
+
+      text =
+        String.split(
+          """
+          defmodule MyApp do
+            def to_list(map) do
+              Foo.Bar.to_list(map)
+            end
+
+            def bar, do: :bar
+          end
+          """,
+          "\n"
+        )
+
+      expected_edit =
+        String.trim("""
+        defmodule MyApp do
+          alias Foo.Bar
+
+          def to_list(map) do
+            Bar.to_list(map)
+          end
+
+          def bar do
```

Comment:
Not sure why Macro.to_string picks to do/end block the one-liner. The ast from a one liner and a do/end block is the same so that's probably why. I might be missing some context data in one of the cases when testing in iex.